Shared/Samples/Authenticate with OAuth/AuthenticateWithOAuthView.swift

```diff
@@ -53,24 +53,22 @@ struct AuthenticateWithOAuthView: View {
                 // keychain and remove `signOut()` from `onDisappear`.
                 // setupPersistentCredentialStorage()
             }
-            .onDisappear {
+            .onTeardown {
                 // Resetting the challenge handlers and clearing credentials here in `onDisappear`
                 // so user is prompted to enter credentials every time trying the sample. In real
                 // world applications, do these from sign out functionality of the application.
                 
                 // Resets challenge handlers.
                 ArcGISEnvironment.authenticationManager.handleChallenges(using: nil)
                 
-                signOut()
+                await signOut()
             }
     }
     
     /// Signs out from the portal by revoking OAuth tokens and clearing credential stores.
-    private func signOut() {
-        Task {
-            await ArcGISEnvironment.authenticationManager.revokeOAuthTokens()
-            await ArcGISEnvironment.authenticationManager.clearCredentialStores()
-        }
+    private func signOut() async {
+        await ArcGISEnvironment.authenticationManager.revokeOAuthTokens()
+        await ArcGISEnvironment.authenticationManager.clearCredentialStores()
```

Comment:
These are functions of extension in the toolkit. The new `signOut` function is exposed. Should all samples use it?

Reply:
> Also, I notice that the AuthenticationManager.signOut() method is missing a Since tag in the toolkit. Does it need one?

Yes, good to add.